### PR TITLE
feat(apprunner): add ipAddressType property to the Service class

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

N/A

### Reason for this change
AppRunner supported Dual Stack.

https://aws.amazon.com/about-aws/whats-new/2023/11/aws-app-runner-supports-ipv6-public-inbound-traffic/?nc1=h_ls

But current L2 Construct (alpha module) does not support it.

### Description of changes
Add ipAddressType property to the Service class

### Description of how you validated changes
Add unit tests and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts [hunk 1]
```diff
@@ -723,6 +723,13 @@ export interface ServiceProps {
    * @default - Use an AWS managed key
    */
   readonly kmsKey?: kms.IKey;
+
+  /**
+   * The IP address type for your incoming public network configuration.
+   *
+   * @default - IPV4
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts:730]:
```suggestion
   * @default - IpAddressType.IPV4
```

File: packages/@aws-cdk/aws-apprunner-alpha/README.md [hunk 2]
```diff
@@ -180,6 +180,20 @@ new apprunner.Service(this, 'Service', {
...
+```ts
+new apprunner.Service(this, 'Service', {
+  source: apprunner.Source.fromEcrPublic({
+    imageConfiguration: { port: 8000 },
+    imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
+  }),
+  ipAddressType: apprunner.IpAddressType.DUAL_STACK,
+});
+```
+
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-apprunner-alpha/README.md:196]:
```suggestion

**Note**: Currently, App Runner supports dual stack for only Public endpoint.
Only IPv4 is supported for Private endpoint.
If you update a service that's using dual-stack Public endpoint to a Private endpoint,
your App Runner service will default to support only IPv4 for Private endpoint and fail
to receive traffic originating from IPv6 endpoint.

```
Might be worth highlighting ([docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apprunner-service-networkconfiguration.html#cfn-apprunner-service-networkconfiguration-ipaddresstype)).

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Comment by mazyu36** [comment]
@lpizzinidev 
Thank you for the review !
I've addressed all your comments.

**Review by aaythapa — APPROVED** [review]

**Pull request merged**